test/HydratorDeprecationTest.php

```diff
@@ -23,4 +23,15 @@ public function testPassingHydratorExtendingStdlibAbstractHydratorToTypehintedMe
         $hydratorInjected->setHydrator($hydrator);
         $this->assertSame($hydrator, $hydratorInjected->hydrator);
     }
+
+    public function testDeprecatedHydratorInterfaceIsAcceptedByMethodsTypehintedWithNewInterface()
+    {
+        $hydratorInjected = new TestAsset\HydratorInjectedObjectUsingDeprecatedInterfaceHydrator();
```

Comment:
It would reduce the number of TestAsset classes added by this PR, that's for sure.  I'll take a stab at it this morning.